test/test_nn.py

```diff
@@ -4463,6 +4463,17 @@ def pad(tensor, length):
         with self.assertRaisesRegex(RuntimeError, 'You can pass `enforce_sorted=False`'):
             packed = rnn_utils.pack_padded_sequence(torch.randn(3, 3), [1, 3, 2])
 
+    def test_pack_padded_sequence_default_cuda_tensor(self):
+        # Set default tensor type to cuda.FloatTensor, as this has triggered the error
+        torch.set_default_tensor_type(torch.cuda.FloatTensor)
```

Comment:
Tests are probably failing because this function is being run unconditionally, even when there is no CUDA available

Reply:
Thanks for the review!
Tests are still failing. Is the decorator `@unittest.skipIf(not TEST_CUDA, 'CUDA not available')` not enough to skip this test on CPU machines?

Reply:
are you sure it's failing on CPU-only machines?  I checked some cases and they were failing on cuda.

Reply:
You are right. The comment was hastily.
This seems to be one reasons for failing tests:
`../.local/lib/python2.7/site-packages/caffe2/python/operator_test/layer_norm_op_test.py::TestLayerNormOp::test_layer_norm_op_c10 FAILED [ 84%]`

However, other tests fail with this output and I think this one might be more relevant:
```
Feb 07 08:38:34 test_pack_padded_sequence (__main__.TestNN) ... ok
Feb 07 08:38:35 test_pack_padded_sequence_default_cuda_tensor (__main__.TestNN) ... ok
Feb 07 08:38:35 test_pack_sequence (__main__.TestNN) ... ok
Feb 07 08:38:35 test_pad (__main__.TestNN) ... /opt/conda/lib/python3.6/site-packages/torch/autograd/gradcheck.py:217: UserWarning: At least one of the inputs that requires gradient is not of double precision floating point. This check will likely fail if all the inputs are not of double precision floating point. 
Feb 07 08:38:35   'At least one of the inputs that requires gradient '
Feb 07 08:38:35 ERROR
Feb 07 08:38:35 test_pad_scalar_error (__main__.TestNN) ... ok
Feb 07 08:38:35 test_pad_sequence (__main__.TestNN) ... ok
Feb 07 08:38:35 test_pairwise_distance (__main__.TestNN) ... ERROR
```

Could it be that resetting the default tensor type to `torch.FloatTensor` could trigger these issues?
I'll try to debug it on my machine further.